### PR TITLE
feat(SystemViews): Add Workload filter

### PR DESCRIPTION
**Pull request by computercamplove**
## Jira
[RHINENG-25379](https://redhat.atlassian.net/browse/RHINENG-25379)

## What
New Workload filter
depends on https://github.com/RedHatInsights/insights-host-inventory/pull/3839

## Why
to support use cases currently relying on the Global Filter, ahead of its decommissioning.

## Testing

1. enable SystemsView by running localStorage.setItem("ui-inventory-views", "true")
2. go to Inventory / Systems
3. Select Workload from filters dropdown
4. try stuff (for example after filtering by SAP check system details Workload filed)

## Screenshots/Videos (if applicable)
image
image

## Summary by Sourcery

Add workload-based filtering to SystemsView and wire it into the systems query.

New Features:
- Introduce a Workload checkbox filter in SystemsView for selecting systems by workloads such as Ansible, SAP, and databases.

Enhancements:
- Extend the systems query to translate selected workloads into system_profile.workloads presence filters for the host list API.

Tests:
- Add unit tests for the workloads filter utility to validate mapping logic, handling of empty input, and key order preservation.

[RHINENG-25379]: https://redhat.atlassian.net/browse/RHINENG-25379?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by computercamplove** [comment]
> * In `fetchSystems`, the new `workloads` filter is added as a separate `filter` object, which will overwrite any previously constructed `filter` in the request body; instead, merge `workloads` into the existing `filter.system_profile` structure so all filters can be applied together.

this comment not relevant anymore

**Comment by computercamplove** [comment]
code is ready for review
- currently filter works as AND when we apply multiple workloads, i am waiting for approval in the backend change https://github.com/RedHatInsights/insights-host-inventory/pull/3839

File: src/components/SystemsView/utils/workloadsFilter.test.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  return Object.fromEntries(keys.map((key) => [key, NOT_NIL]));
+}
+
+describe('buildWorkloadsFilter', () => {
+  it('returns undefined for empty or undefined input', () => {
+    expect(buildWorkloadsFilter(undefined)).toBeUndefined();
+    expect(buildWorkloadsFilter([])).toBeUndefined();
+  });
+
+  it('maps each selected key to { is: not_nil } (presence on system_profile.workloads.<key>)', () => {
```
Comment (raswonders) [inline src/components/SystemsView/utils/workloadsFilter.test.ts]: One of these tests seems redundant to me, we check subset of keys first, and in next test we test all keys with same condition. 

If we test allKeys only we should have same guarantees but less code...

**Review by raswonders — APPROVED** [review]
Good job, UI works as expected, will double-check URL params in PR fixing them for all filters. Small nit pick about one test buts over all looks good.

File: src/components/SystemsView/utils/workloadsFilter.test.ts (hunk 1 above)
Comment (computercamplove) [reply src/components/SystemsView/utils/workloadsFilter.test.ts]: true, let me remove it
